src/protocols/secure_channel/MessageCounterManager.h

```diff
@@ -24,25 +24,36 @@
 
 #include <protocols/Protocols.h>
 #include <system/SystemPacketBuffer.h>
+#include <transport/MessageCounterManagerInterface.h>
 
 namespace chip {
-namespace Messaging {
-
-constexpr uint16_t kMsgCounterChallengeSize   = 8;   // The size of the message counter synchronization request message.
-constexpr uint16_t kMsgCounterSyncRespMsgSize = 12;  // The size of the message counter synchronization response message.
-constexpr uint32_t kMsgCounterSyncTimeout     = 500; // The amount of time(in milliseconds) which a peer is given to respond
-                                                     // to a message counter synchronization request.
+namespace secure_channel {
 
 class ExchangeManager;
 
-class MessageCounterSyncMgr : public Messaging::ExchangeDelegate
+class MessageCounterManager : public Messaging::ExchangeDelegate, public Transport::MessageCounterManagerInterface
 {
 public:
-    MessageCounterSyncMgr() : mExchangeMgr(nullptr) {}
+    static constexpr uint16_t kChallengeSize   = Transport::PeerMessageCounter::kChallengeSize;
+    static constexpr uint16_t kCounterSize     = 4;
+    static constexpr uint16_t kSyncRespMsgSize = kChallengeSize + kCounterSize;
+    static constexpr uint16_t kMacSize         = 16;
```

Comment:
Please, no.  We have a constant for this already (`kMaxTagLen` or `kMaxFooterSize` depending on which code you're in and the semantics you care about) and adding another hardcoded constant is just asking for them to get out of sync.